FF1Lib/ItemLists.cs

```diff
@@ -160,59 +160,219 @@ public static class ItemLists
 
 		public static List<Item> GetIncentivePool(Flags flags)
 		{
-			var incentivePool = new List<Item>();
-
-			if (flags.IncentivizeMasamune ?? false) incentivePool.Add(Item.Masamune);
-			if (flags.IncentivizeKatana ?? false) incentivePool.Add(Item.Katana);
-			if (flags.IncentivizeVorpal ?? false) incentivePool.Add(Item.Vorpal);
-			if (flags.IncentivizeDefCastWeapon ?? false) incentivePool.Add(Item.Defense);
-			if (flags.IncentivizeOffCastWeapon ?? false) incentivePool.Add(Item.ThorHammer);
-			if (flags.IncentivizeOpal ?? false) incentivePool.Add(Item.Opal);
-			if (flags.IncentivizeOtherCastArmor ?? false) incentivePool.Add(Item.PowerGauntlets);
-			if (flags.IncentivizeDefCastArmor ?? false) incentivePool.Add(Item.WhiteShirt);
-			if (flags.IncentivizeOffCastArmor ?? false) incentivePool.Add(Item.BlackShirt);
-			if (flags.IncentivizeRibbon ?? false) incentivePool.Add(Item.Ribbon);
-			if (flags.IncentivizeSlab ?? false) incentivePool.Add(Item.Slab);
-			if (flags.IncentivizeRuby ?? false) incentivePool.Add(Item.Ruby);
-			if (flags.IncentivizeFloater ?? false) incentivePool.Add(Item.Floater);
-			if (flags.IncentivizeTnt ?? false) incentivePool.Add(Item.Tnt);
-			if (flags.IncentivizeCrown ?? false) incentivePool.Add(Item.Crown);
-			if (flags.IncentivizeTail ?? false) incentivePool.Add(Item.Tail);
-			if (flags.IncentivizeAdamant ?? false) incentivePool.Add(Item.Adamant);
-
-			if (flags.IncentivizeBridge) incentivePool.Add(Item.Bridge);
-			if (flags.IncentivizeLute ?? false) incentivePool.Add(Item.Lute);
-			if (flags.IncentivizeShip ?? false) incentivePool.Add(Item.Ship);
-			if (flags.IncentivizeRod ?? false) incentivePool.Add(Item.Rod);
-			if (flags.IncentivizeCanoe ?? false) incentivePool.Add(Item.Canoe);
-			if (flags.IncentivizeCube ?? false) incentivePool.Add(Item.Cube);
-			if (flags.IncentivizeBottle ?? false) incentivePool.Add(Item.Bottle);
-
-			if (flags.IncentivizeKey ?? false) incentivePool.Add(Item.Key);
-			if (flags.IncentivizeCrystal ?? false) incentivePool.Add(Item.Crystal);
-			if (flags.IncentivizeOxyale ?? false) incentivePool.Add(Item.Oxyale);
-			if (flags.IncentivizeCanal ?? false) incentivePool.Add(Item.Canal);
-			if (flags.IncentivizeHerb ?? false) incentivePool.Add(Item.Herb);
-			if (flags.IncentivizeChime ?? false) incentivePool.Add(Item.Chime);
-			if (flags.IncentivizeXcalber ?? false) incentivePool.Add(Item.Xcalber);
-
-			return incentivePool.Concat(ItemLists.AllQuestItems).Distinct().ToList();
+			List<Item> incentivePool = new();
+
+			if (flags.IncentivizeMasamune ?? false)

```

Comment:
this might be the only thing that's really bothering me, as this change lowers readability imo, but this might be also a sign that these flags should be handled differently

Reply:
I would have hoped that it would at least make one-liners to make the whole thing a little more compact.

Reply:
I agree that this makes it less readable, though in most cases cramming as much into one line as possible isnt great imo, so as a general rule I think this is good, but this function likely needs exceptions.

Reply:
Make a list that maps flags to items and iterate over the list with a simple function.